custom/nic_compliance/forms.py

```diff
@@ -0,0 +1,39 @@
+from django import forms
+from django.utils.translation import ugettext as _
+from django.core.exceptions import ValidationError
+from django.contrib.auth import password_validation
+
+
+class EncodedPasswordChangeForm(object):
+    def clean_new_password1(self):
+        from corehq.apps.domain.forms import clean_password
+        from corehq.apps.hqwebapp.utils import decode_password
+        new_password = decode_password(self.cleaned_data.get('new_password1'))
+        # User might not be able to submit empty password but decode_password might
+        # return empty password in case the password hashing is messed up with
+        if new_password == '':
+            raise ValidationError(
+                _("Password cannot be empty"), code='new_password1_empty',
+            )
+
+        return clean_password(new_password)
+
+    def clean_new_password2(self):
+        from corehq.apps.hqwebapp.utils import decode_password
+        password2 = decode_password(self.cleaned_data.get('new_password2'))
+        if password2 == '':
+            raise ValidationError(
+                _("Password cannot be empty"), code='new_password2_empty',
+            )
+
+        password1 = self.cleaned_data.get('new_password1')
```

Comment:
Seems like `clean_new_password1` would be called before `clean_new_password2`. 
```
    def _clean_fields(self):
        for name, field in self.fields.items():
```
This is the snippet from django forms and that fields is an ordered list and when adding the fields "all fields are kept in the order defined in the class" since we are not specifying any order

More details [here](https://github.com/django/django/blob/1.10/django/forms/forms.py#L93)